src/liboslexec/aligned_allocator.h

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// https://github.com/imageworks/OpenShadingLanguage
+
+/* NOTE:  Can be replaced with boost 1.56 boost::alligned_allocator
+ * Current boost dependency is only boost 1.55, so consider
+ * removing this when our boost minimum becomes >= 1.56.
+ * No expected performance difference, just one less file
+ * NOTE: pointer_or_number and is_aligned appear useful and should be
+ * pulled into their own files
+ */
+#pragma once
+
+#include <limits>
+#include <type_traits>
+#include <utility>
+
+#include <OSL/oslversion.h>
+
+OSL_NAMESPACE_ENTER
+
+namespace pvt {
+
+// For converting pointers to numbers for alignment or other math operations
+template<typename T = void>
+union pointer_or_number {
+    pointer_or_number() {}
+    pointer_or_number(const T * pointer_)
+    :pointer(pointer_)
+    {}
+    pointer_or_number(size_t number_)
+    :number(number_)
+    {}
+
+    const T * pointer;
+    size_t number;
+};
+
+
+
+template <size_t ByteAlignmentT, typename T>
+inline bool is_aligned(const T *a_pointer)
+{
+    pointer_or_number<T> pon;
+    pon.pointer = a_pointer;
+    return (pon.number%ByteAlignmentT==0);
+}
+
+
+
+template <class ObjT, size_t Alignment=64>
+class aligned_allocator
+{
+    template <class> friend class aligned_allocator;
+
+    static constexpr int boundary = Alignment;
+public:
+    typedef ObjT value_type;
+    typedef ObjT* pointer;
+    typedef const ObjT* const_pointer;
+    typedef ObjT& reference;
+    typedef const ObjT& const_reference;
+    typedef std::size_t size_type;
+    typedef std::ptrdiff_t difference_type;
+    typedef std::true_type propagate_on_container_move_assignment;
+    typedef std::true_type is_always_equal;
+
+    template< class OtherObjT >
+    struct rebind {
+        typedef aligned_allocator<OtherObjT> other;
+    };
+
+    pointer address( reference a_ref) const;
+    const_pointer address( const_reference a_ref) const;
+
+    pointer allocate(size_type a_count, std::allocator<void>::const_pointer a_hint = 0);
+    void deallocate(pointer a_pointer, size_type a_count);
+
+    size_type max_size() const;
+
+    void construct(pointer a_pointer, const_reference a_value);
+
+    void destroy(pointer a_pointer);
+
+    // Required for C++11 compliance
+    template< class OtherObjT, class... ArgsT >
+    void construct(OtherObjT* a_pointer, ArgsT&&... args);
+    template< class OtherObjT >
+    void destroy(OtherObjT* a_pointer);
+};
+
+// Implementation
+
+
+template <class ObjT>
+typename aligned_allocator<ObjT>::pointer
+aligned_allocator<ObjT>::address( reference a_ref) const
+{
+    return &a_ref;
+}
+
+template <class ObjT>
+typename aligned_allocator<ObjT>::const_pointer
+aligned_allocator<ObjT>::address( const_reference a_ref) const
+{
+    return &a_ref;
+}
+
+template <class ObjT>
+typename aligned_allocator<ObjT>::pointer
+aligned_allocator<ObjT>::allocate(size_type a_count, std::allocator<void>::const_pointer /*a_hint*/)
```

Comment:
On *nix platforms you can just call `posix_memalign` which works with plain old `free`.

On windows you can do `_aligned_malloc` and `_aligned_free`.